### PR TITLE
Sns type filtering

### PR DESCRIPTION
**Pull request by mstrasinskis**
# Motivation

Update `snsFilteredProposalsStore` logic to make sns  proposal `types` filter work.

# Changes

- update `isSns(Native|Generic)NervousSystemFunction` to rely on `MIN_VALID_SNS_GENERIC_NERVOUS_SYSTEM_FUNCTION_ID`. To have the same same logic as `isSnsGenericNervousSystemTypeProposal`.
- `isSnsGenericNervousSystemTypeProposal` util
- update `snsFilteredProposalsStore` logic to filter by types.

# Tests

- add `action` param to `createSnsProposal` mock
- `isGenericNervousSystemTypeProposal`
- `loadSnsProposals` provides `excludeType` parameter
- `SnsProposals` component has working types filtering
- `snsFilteredProposalsStore` should filter by types

# Todos

- [ ] Add entry to changelog (if necessary).
Not yet

### CODE REVIEW

File: frontend/src/lib/components/sns-proposals/SnsProposalsFilters.svelte [hunk 1]
```diff
@@ -28,6 +45,16 @@
 </script>
 
 <FiltersWrapper>
+  <FiltersButton
+    testId="filters-by-types"
+    totalFilters={filtersStore?.types.length ?? 0}
+    activeFilters={filtersStore?.types.filter(({ checked }) => checked)
+      .length ?? 0}
+    on:nnsFilter={() => openFilters("types")}
+    showSpinner={filtersStore?.types.length === 0}
```
Comment (mstrasinskis) [inline frontend/src/lib/components/sns-proposals/SnsProposalsFilters.svelte:54]: test

File: frontend/src/lib/utils/sns-proposals.utils.ts [hunk 2]
```diff
@@ -475,6 +482,83 @@ export const getUniversalProposalStatus = (
...
+    nsFunctions.filter(isGenericNervousSystemFunction).length > 0;
+  const filterableIds = hasGenericNsFunctions
+    ? [...nativeNsFunctionIds, ALL_GENERIC_PROPOSAL_TYPES_ID]
+    : nativeNsFunctionIds;
+  const mapToFilterData = (id: string): Filter<SnsProposalTypeFilterData> => ({
+    id,
+    value: id as keyof typeof i18nKeys.sns_types,
+    name:
+      i18nKeys.sns_types[id as keyof typeof i18nKeys.sns_types] ??
+      // TODO(max): first use native names as fallback
```
Comment (lmuntaner) [inline frontend/src/lib/utils/sns-proposals.utils.ts:550]: Thanks!

File: frontend/src/tests/lib/derived/sns/sns-filtered-proposals.derived.spec.ts [hunk 3]
```diff
@@ -82,13 +84,10 @@ describe("snsFilteredProposalsStore", () => {
       decisionStatus,
     });
 
-    expect(
-      get(snsFilteredProposalsStore)[rootCanisterId.toText()].proposals
-    ).toHaveLength(proposals.length);
+    expect(getProposals()).toHaveLength(proposals.length);
```
Comment (mstrasinskis) [inline frontend/src/tests/lib/derived/sns/sns-filtered-proposals.derived.spec.ts:87]: Just a bit of refactoring to get rid of repetitive piece of code.

**Comment by mstrasinskis** [comment]
@lmuntaner, this is the final part of the `types` filtering feature.

File: frontend/src/tests/lib/derived/sns/sns-filtered-proposals.derived.spec.ts [hunk 4]
```diff
@@ -190,8 +187,131 @@ describe("snsFilteredProposalsStore", () => {
...
+        {
+          id: `${nativeNsFunctionId}`,
+          value: `${nativeNsFunctionId}`,
+          name: "Motion",
+          checked: true,
+        },
+        {
+          id: ALL_SNS_GENERIC_PROPOSAL_TYPES_ID,
+          value: ALL_SNS_GENERIC_PROPOSAL_TYPES_ID,
+          name: "Add a Node",
```
[the remark this answers is not in the document]
Comment (mstrasinskis) [reply frontend/src/tests/lib/derived/sns/sns-filtered-proposals.derived.spec.ts]: Good catch, updated. And thanks a lot for your quick review.

**Pull request merged**